crates/workspace/src/pane.rs

```diff
@@ -1761,30 +1774,64 @@ impl Pane {
-                                        let parent = if entry.is_symlink {
-                                            abs_path.canonicalize().ok()?
-                                        } else {
-                                            abs_path
-                                        }
-                                        .parent()?
-                                        .to_path_buf();
-                                        Some(parent)
+                                        let rel_path = &entry.path;
+                                        let abs_path = (|| {
+                                            let abs_path = worktree.absolutize(rel_path).ok()?;
+                                            let parent = if entry.is_symlink {
+                                                abs_path.canonicalize().ok()?
+                                            } else {
+                                                abs_path
+                                            }
+                                            .to_path_buf();
+                                            Some(parent)
+                                        })();
+                                        Some((Arc::clone(rel_path), abs_path))
                                     })
                                 })
                             })
                             .ok()
                             .flatten();
 
+                        let (rel_path, abs_path) =
+                            paths.map_or((None, None), |(rel, abs)| (Some(rel), abs));
+
+                        let parent_abs_path = abs_path
+                            .as_deref()
+                            .and_then(|path| path.parent())
+                            .map(|path| path.to_path_buf());
+
                         let entry_id = entry.to_proto();
                         menu = menu
+                            .separator()
+                            .when_some(abs_path, |menu, path| {
+                                menu.entry(
+                                    "Copy Path",
+                                    Some(Box::new(CopyPath)),
+                                    cx.handler_for(&pane, move |pane, cx| {
+                                        pane.copy_path(&CopyPath, path.to_path_buf(), cx);
+                                    }),
+                                )
+                            })
+                            .when_some(rel_path, |menu, path| {
+                                menu.entry(
+                                    "Copy Relative Path",
+                                    Some(Box::new(CopyRelativePath)),
+                                    cx.handler_for(&pane, move |pane, cx| {
+                                        pane.copy_relative_path(
+                                            &CopyRelativePath,
+                                            path.to_path_buf(),
```

Comment:
> to_path_buf

This is an extra allocation for nothing, as later all we do with it is `.to_string_lossy().to_string()`.
A `&Path` or a `&PathBuf` is more than just enough.

Reply:
Removing both methods and calling clipboard inline resolves this